src/h5web/toolbar/controls/DomainSlider/hooks.ts

```diff
@@ -1,18 +1,18 @@
 import { useEffect, useState } from 'react';
+import { createMemo } from 'react-use';
 import type { Domain } from '../../../../packages/lib';
-import type { CustomDomain } from '../../../vis-packs/core/models';
-import { getSliderDomain } from './utils';
+import { getAppliedDomain, getSliderScale } from './utils';
 
-export function useAppliedDomain(
-  dataDomain: Domain,
-  customDomain: CustomDomain
-) {
-  const state = useState(getSliderDomain(dataDomain, customDomain));
-  const [, setAppliedDomain] = state;
+export const useAppliedDomain = createMemo(getAppliedDomain);
+export const useSliderScale = createMemo(getSliderScale);
+
+export function useSliderDomain(appliedDomain: Domain) {
```

Comment:
Wild idea: do you think `useSliderScale` and `useSliderDomain` could be merged in a single hook ?

Reply:
Hmm I don't see the benefit of squeezing those two hooks into one, sorry. Seems nice that `useSliderDomain` matches the signature of `useState`.

Reply:
Also, `useSliderScale` will ow be called in `ScaledSlider`, whereas `useSliderDomain` will stay in `DomainSlider`.

Reply:
No problem, I just thought it was worth mentioning it